cli/pkg/app/read.go

```diff
@@ -0,0 +1,64 @@
+// Copyright © 2020 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// httpRead reads data from a remote file using URL
+func httpRead(url string) ([]byte, int, error) {
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, 0, err
+	}
+	defer resp.Body.Close()
+
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	return data, resp.StatusCode, nil
+}
+
+// FetchHubResource will fetch resource from Hub and depending on status code
+// returns the data
+func FetchHubResource(stream *Stream, cli CLI, api string) ([]byte, bool, error) {
+
+	url := cli.API() + api
+	data, status, err := httpRead(url)
+	if err != nil {
+		fmt.Fprintln(stream.Err, "Error: failed to read resource", err)
+		return nil, false, nil
+	}
+
+	switch status {
+	case 200:
+		return data, true, nil
+	case 404:
+		fmt.Fprintln(stream.Out, "No Resource Found")
```

Comment:
this should return an error and let the caller print or ignore instead of printing it here .. The rule this function breaks is SRP ... 
It not only fetches the resource (its responsibility)  but also prints it (another responsibility)